format.go

```diff
@@ -579,3 +591,123 @@ func (d *digits) round(prec int) {
 		d.ndig = prec
 	}
 }
+
+// FmtFormat stores a formatting layout for Decimal values.
+type FmtFormat struct {
```

Comment:
Another alternative here would be to declare `FmtFormat` as an interface implementing `Flag()`, `Width()` and `Precision()` (same signature as fmt.State) and have a private implementation.